tests/mappers/test_exon_genomic_coords.py

```diff
@@ -19,286 +20,533 @@ def test_egc_mapper(test_cool_seq_tool):
+            alt_end_i=80487782,
+            alt_strand=Strand.NEGATIVE,
+        ),
+        ExonCoord(
+            ord=1,
+            tx_start_i=1557,
+            tx_end_i=2446,
+            alt_start_i=80499493,
+            alt_end_i=80500382,
+            alt_strand=Strand.NEGATIVE,
+        ),
+        ExonCoord(
+            ord=2,
+            tx_start_i=2446,
+            tx_end_i=2545,
+            alt_start_i=80513909,
+            alt_end_i=80514008,
+            alt_strand=Strand.NEGATIVE,
+        ),
+        ExonCoord(
+            ord=3,
+            tx_start_i=2545,
+            tx_end_i=2722,
+            alt_start_i=80518402,
+            alt_end_i=80518579,
+            alt_strand=Strand.NEGATIVE,
+        ),
+        ExonCoord(
+            ord=4,
+            tx_start_i=2722,
+            tx_end_i=2895,
+            alt_start_i=80518781,
+            alt_end_i=80518954,
+            alt_strand=Strand.NEGATIVE,
+        ),
+        ExonCoord(
+            ord=5,
+            tx_start_i=2895,
+            tx_end_i=9938,
+            alt_start_i=80519222,
+            alt_end_i=80526265,
+            alt_strand=Strand.NEGATIVE,
+        ),
+    ]
+
+
 @pytest.fixture(scope="module")
 def tpm3_exon1():
-    """Create test fixture for TPM3 exon 1."""
+    """Create test fixture for TPM3 exon 1 (negative strand)."""
     params = {
-        "chr": "NC_000001.11",
         "gene": "TPM3",
-        "pos": 154192134,
-        "exon": 1,
-        "exon_offset": 0,
-        "transcript": "NM_152263.3",
-        "strand": Strand.NEGATIVE,
+        "genomic_accession": "NC_000001.11",
+        "tx_ac": "NM_152263.3",
+        "seg": {
+            "exon_ord": 0,
+            "offset": 0,
+            "genomic_location": {
+                "type": "SequenceLocation",
+                "sequenceReference": {
+                    "type": "SequenceReference",
+                    "refgetAccession": "SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO",
+                },
+                "end": 154191901,
+            },
+        },
     }
-    return TranscriptExonData(**params)
+    return _GenomicTxSeg(**params)
 
 
 @pytest.fixture(scope="module")
 def tpm3_exon8():
-    """Create test fixture for TPM3 exon 8."""
+    """Create test fixture for TPM3 exon 8 (negative strand)."""
     params = {
-        "chr": "NC_000001.11",
         "gene": "TPM3",
-        "pos": 154170400,
-        "exon": 8,
-        "exon_offset": 0,
-        "transcript": "NM_152263.3",
-        "strand": Strand.NEGATIVE,
+        "genomic_accession": "NC_000001.11",
+        "tx_ac": "NM_152263.3",
+        "seg": {
+            "exon_ord": 7,
+            "offset": 0,
+            "genomic_location": {
+                "type": "SequenceLocation",
+                "sequenceReference": {
+                    "type": "SequenceReference",
+                    "refgetAccession": "SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO",
+                },
+                "start": 154170469,
```

Comment:
I'm not sure about this value - I think this should be 154170399,

Reply:
Same comment as above